crates/vm/src/arch/integration_api.rs

```diff
@@ -360,6 +359,21 @@ where
     fn width(&self) -> usize {
         self.adapter.width() + self.core.width()
     }
+
+    fn columns(&self) -> Vec<String> {
```

Comment:
Added this extra step so that later we can be able to distinguish where each column comes from